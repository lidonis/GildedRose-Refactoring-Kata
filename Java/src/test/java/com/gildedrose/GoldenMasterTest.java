package com.gildedrose;

import org.approvaltests.combinations.CombinationApprovals;
import org.approvaltests.combinations.SkipCombination;
import org.junit.jupiter.api.Test;

public class GoldenMasterTest {

  private static final String SULFURAS = "Sulfuras, Hand of Ragnaros";

  @Test
  public void testUpdate() {
    String[] itemNames = {"+5 Dexterity Vest", "Aged Brie", "Elixir of the Mongoose",
        SULFURAS, "Backstage passes to a TAFKAL80ETC concert",
        "Conjured Mana Cake"};
    Integer[] sellIns = {-5, -1, 0, 2, 5, 10, 15};
    Integer[] qualities = {0, 1, 7, 20, 49, 50, 80};
    CombinationApprovals.verifyAllCombinations(this::updateItem,
        itemNames, sellIns, qualities);
  }

  private Object updateItem(String name, int sellIn, int quality) {
    if ((SULFURAS.equals(name) && (quality != 80)) || (!SULFURAS.equals(name) && (quality == 80))) {
      throw new SkipCombination();
    }

    Item[] items = new Item[]{new Item(name, sellIn, quality)};
    GildedRose gildedRose = new GildedRose(items);
    gildedRose.updateQuality();
    return items[0];
  }

}
