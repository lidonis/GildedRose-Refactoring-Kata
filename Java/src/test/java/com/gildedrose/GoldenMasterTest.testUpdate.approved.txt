[+5 Dexterity Vest, -5, 0] => +5 Dexterity Vest, -6, 0 
[+5 Dexterity Vest, -5, 1] => +5 Dexterity Vest, -6, 0 
[+5 Dexterity Vest, -5, 7] => +5 Dexterity Vest, -6, 5 
[+5 Dexterity Vest, -5, 20] => +5 Dexterity Vest, -6, 18 
[+5 Dexterity Vest, -5, 49] => +5 Dexterity Vest, -6, 47 
[+5 Dexterity Vest, -5, 50] => +5 Dexterity Vest, -6, 48 
[+5 Dexterity Vest, -1, 0] => +5 Dexterity Vest, -2, 0 
[+5 Dexterity Vest, -1, 1] => +5 Dexterity Vest, -2, 0 
[+5 Dexterity Vest, -1, 7] => +5 Dexterity Vest, -2, 5 
[+5 Dexterity Vest, -1, 20] => +5 Dexterity Vest, -2, 18 
[+5 Dexterity Vest, -1, 49] => +5 Dexterity Vest, -2, 47 
[+5 Dexterity Vest, -1, 50] => +5 Dexterity Vest, -2, 48 
[+5 Dexterity Vest, 0, 0] => +5 Dexterity Vest, -1, 0 
[+5 Dexterity Vest, 0, 1] => +5 Dexterity Vest, -1, 0 
[+5 Dexterity Vest, 0, 7] => +5 Dexterity Vest, -1, 5 
[+5 Dexterity Vest, 0, 20] => +5 Dexterity Vest, -1, 18 
[+5 Dexterity Vest, 0, 49] => +5 Dexterity Vest, -1, 47 
[+5 Dexterity Vest, 0, 50] => +5 Dexterity Vest, -1, 48 
[+5 Dexterity Vest, 2, 0] => +5 Dexterity Vest, 1, 0 
[+5 Dexterity Vest, 2, 1] => +5 Dexterity Vest, 1, 0 
[+5 Dexterity Vest, 2, 7] => +5 Dexterity Vest, 1, 6 
[+5 Dexterity Vest, 2, 20] => +5 Dexterity Vest, 1, 19 
[+5 Dexterity Vest, 2, 49] => +5 Dexterity Vest, 1, 48 
[+5 Dexterity Vest, 2, 50] => +5 Dexterity Vest, 1, 49 
[+5 Dexterity Vest, 5, 0] => +5 Dexterity Vest, 4, 0 
[+5 Dexterity Vest, 5, 1] => +5 Dexterity Vest, 4, 0 
[+5 Dexterity Vest, 5, 7] => +5 Dexterity Vest, 4, 6 
[+5 Dexterity Vest, 5, 20] => +5 Dexterity Vest, 4, 19 
[+5 Dexterity Vest, 5, 49] => +5 Dexterity Vest, 4, 48 
[+5 Dexterity Vest, 5, 50] => +5 Dexterity Vest, 4, 49 
[+5 Dexterity Vest, 10, 0] => +5 Dexterity Vest, 9, 0 
[+5 Dexterity Vest, 10, 1] => +5 Dexterity Vest, 9, 0 
[+5 Dexterity Vest, 10, 7] => +5 Dexterity Vest, 9, 6 
[+5 Dexterity Vest, 10, 20] => +5 Dexterity Vest, 9, 19 
[+5 Dexterity Vest, 10, 49] => +5 Dexterity Vest, 9, 48 
[+5 Dexterity Vest, 10, 50] => +5 Dexterity Vest, 9, 49 
[+5 Dexterity Vest, 15, 0] => +5 Dexterity Vest, 14, 0 
[+5 Dexterity Vest, 15, 1] => +5 Dexterity Vest, 14, 0 
[+5 Dexterity Vest, 15, 7] => +5 Dexterity Vest, 14, 6 
[+5 Dexterity Vest, 15, 20] => +5 Dexterity Vest, 14, 19 
[+5 Dexterity Vest, 15, 49] => +5 Dexterity Vest, 14, 48 
[+5 Dexterity Vest, 15, 50] => +5 Dexterity Vest, 14, 49 
[Aged Brie, -5, 0] => Aged Brie, -6, 2 
[Aged Brie, -5, 1] => Aged Brie, -6, 3 
[Aged Brie, -5, 7] => Aged Brie, -6, 9 
[Aged Brie, -5, 20] => Aged Brie, -6, 22 
[Aged Brie, -5, 49] => Aged Brie, -6, 50 
[Aged Brie, -5, 50] => Aged Brie, -6, 50 
[Aged Brie, -1, 0] => Aged Brie, -2, 2 
[Aged Brie, -1, 1] => Aged Brie, -2, 3 
[Aged Brie, -1, 7] => Aged Brie, -2, 9 
[Aged Brie, -1, 20] => Aged Brie, -2, 22 
[Aged Brie, -1, 49] => Aged Brie, -2, 50 
[Aged Brie, -1, 50] => Aged Brie, -2, 50 
[Aged Brie, 0, 0] => Aged Brie, -1, 2 
[Aged Brie, 0, 1] => Aged Brie, -1, 3 
[Aged Brie, 0, 7] => Aged Brie, -1, 9 
[Aged Brie, 0, 20] => Aged Brie, -1, 22 
[Aged Brie, 0, 49] => Aged Brie, -1, 50 
[Aged Brie, 0, 50] => Aged Brie, -1, 50 
[Aged Brie, 2, 0] => Aged Brie, 1, 1 
[Aged Brie, 2, 1] => Aged Brie, 1, 2 
[Aged Brie, 2, 7] => Aged Brie, 1, 8 
[Aged Brie, 2, 20] => Aged Brie, 1, 21 
[Aged Brie, 2, 49] => Aged Brie, 1, 50 
[Aged Brie, 2, 50] => Aged Brie, 1, 50 
[Aged Brie, 5, 0] => Aged Brie, 4, 1 
[Aged Brie, 5, 1] => Aged Brie, 4, 2 
[Aged Brie, 5, 7] => Aged Brie, 4, 8 
[Aged Brie, 5, 20] => Aged Brie, 4, 21 
[Aged Brie, 5, 49] => Aged Brie, 4, 50 
[Aged Brie, 5, 50] => Aged Brie, 4, 50 
[Aged Brie, 10, 0] => Aged Brie, 9, 1 
[Aged Brie, 10, 1] => Aged Brie, 9, 2 
[Aged Brie, 10, 7] => Aged Brie, 9, 8 
[Aged Brie, 10, 20] => Aged Brie, 9, 21 
[Aged Brie, 10, 49] => Aged Brie, 9, 50 
[Aged Brie, 10, 50] => Aged Brie, 9, 50 
[Aged Brie, 15, 0] => Aged Brie, 14, 1 
[Aged Brie, 15, 1] => Aged Brie, 14, 2 
[Aged Brie, 15, 7] => Aged Brie, 14, 8 
[Aged Brie, 15, 20] => Aged Brie, 14, 21 
[Aged Brie, 15, 49] => Aged Brie, 14, 50 
[Aged Brie, 15, 50] => Aged Brie, 14, 50 
[Elixir of the Mongoose, -5, 0] => Elixir of the Mongoose, -6, 0 
[Elixir of the Mongoose, -5, 1] => Elixir of the Mongoose, -6, 0 
[Elixir of the Mongoose, -5, 7] => Elixir of the Mongoose, -6, 5 
[Elixir of the Mongoose, -5, 20] => Elixir of the Mongoose, -6, 18 
[Elixir of the Mongoose, -5, 49] => Elixir of the Mongoose, -6, 47 
[Elixir of the Mongoose, -5, 50] => Elixir of the Mongoose, -6, 48 
[Elixir of the Mongoose, -1, 0] => Elixir of the Mongoose, -2, 0 
[Elixir of the Mongoose, -1, 1] => Elixir of the Mongoose, -2, 0 
[Elixir of the Mongoose, -1, 7] => Elixir of the Mongoose, -2, 5 
[Elixir of the Mongoose, -1, 20] => Elixir of the Mongoose, -2, 18 
[Elixir of the Mongoose, -1, 49] => Elixir of the Mongoose, -2, 47 
[Elixir of the Mongoose, -1, 50] => Elixir of the Mongoose, -2, 48 
[Elixir of the Mongoose, 0, 0] => Elixir of the Mongoose, -1, 0 
[Elixir of the Mongoose, 0, 1] => Elixir of the Mongoose, -1, 0 
[Elixir of the Mongoose, 0, 7] => Elixir of the Mongoose, -1, 5 
[Elixir of the Mongoose, 0, 20] => Elixir of the Mongoose, -1, 18 
[Elixir of the Mongoose, 0, 49] => Elixir of the Mongoose, -1, 47 
[Elixir of the Mongoose, 0, 50] => Elixir of the Mongoose, -1, 48 
[Elixir of the Mongoose, 2, 0] => Elixir of the Mongoose, 1, 0 
[Elixir of the Mongoose, 2, 1] => Elixir of the Mongoose, 1, 0 
[Elixir of the Mongoose, 2, 7] => Elixir of the Mongoose, 1, 6 
[Elixir of the Mongoose, 2, 20] => Elixir of the Mongoose, 1, 19 
[Elixir of the Mongoose, 2, 49] => Elixir of the Mongoose, 1, 48 
[Elixir of the Mongoose, 2, 50] => Elixir of the Mongoose, 1, 49 
[Elixir of the Mongoose, 5, 0] => Elixir of the Mongoose, 4, 0 
[Elixir of the Mongoose, 5, 1] => Elixir of the Mongoose, 4, 0 
[Elixir of the Mongoose, 5, 7] => Elixir of the Mongoose, 4, 6 
[Elixir of the Mongoose, 5, 20] => Elixir of the Mongoose, 4, 19 
[Elixir of the Mongoose, 5, 49] => Elixir of the Mongoose, 4, 48 
[Elixir of the Mongoose, 5, 50] => Elixir of the Mongoose, 4, 49 
[Elixir of the Mongoose, 10, 0] => Elixir of the Mongoose, 9, 0 
[Elixir of the Mongoose, 10, 1] => Elixir of the Mongoose, 9, 0 
[Elixir of the Mongoose, 10, 7] => Elixir of the Mongoose, 9, 6 
[Elixir of the Mongoose, 10, 20] => Elixir of the Mongoose, 9, 19 
[Elixir of the Mongoose, 10, 49] => Elixir of the Mongoose, 9, 48 
[Elixir of the Mongoose, 10, 50] => Elixir of the Mongoose, 9, 49 
[Elixir of the Mongoose, 15, 0] => Elixir of the Mongoose, 14, 0 
[Elixir of the Mongoose, 15, 1] => Elixir of the Mongoose, 14, 0 
[Elixir of the Mongoose, 15, 7] => Elixir of the Mongoose, 14, 6 
[Elixir of the Mongoose, 15, 20] => Elixir of the Mongoose, 14, 19 
[Elixir of the Mongoose, 15, 49] => Elixir of the Mongoose, 14, 48 
[Elixir of the Mongoose, 15, 50] => Elixir of the Mongoose, 14, 49 
[Sulfuras, Hand of Ragnaros, -5, 80] => Sulfuras, Hand of Ragnaros, -5, 80 
[Sulfuras, Hand of Ragnaros, -1, 80] => Sulfuras, Hand of Ragnaros, -1, 80 
[Sulfuras, Hand of Ragnaros, 0, 80] => Sulfuras, Hand of Ragnaros, 0, 80 
[Sulfuras, Hand of Ragnaros, 2, 80] => Sulfuras, Hand of Ragnaros, 2, 80 
[Sulfuras, Hand of Ragnaros, 5, 80] => Sulfuras, Hand of Ragnaros, 5, 80 
[Sulfuras, Hand of Ragnaros, 10, 80] => Sulfuras, Hand of Ragnaros, 10, 80 
[Sulfuras, Hand of Ragnaros, 15, 80] => Sulfuras, Hand of Ragnaros, 15, 80 
[Backstage passes to a TAFKAL80ETC concert, -5, 0] => Backstage passes to a TAFKAL80ETC concert, -6, 0 
[Backstage passes to a TAFKAL80ETC concert, -5, 1] => Backstage passes to a TAFKAL80ETC concert, -6, 0 
[Backstage passes to a TAFKAL80ETC concert, -5, 7] => Backstage passes to a TAFKAL80ETC concert, -6, 0 
[Backstage passes to a TAFKAL80ETC concert, -5, 20] => Backstage passes to a TAFKAL80ETC concert, -6, 0 
[Backstage passes to a TAFKAL80ETC concert, -5, 49] => Backstage passes to a TAFKAL80ETC concert, -6, 0 
[Backstage passes to a TAFKAL80ETC concert, -5, 50] => Backstage passes to a TAFKAL80ETC concert, -6, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 0] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 1] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 7] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 20] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 49] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 50] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 0] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 1] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 7] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 20] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 49] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 50] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 2, 0] => Backstage passes to a TAFKAL80ETC concert, 1, 3 
[Backstage passes to a TAFKAL80ETC concert, 2, 1] => Backstage passes to a TAFKAL80ETC concert, 1, 4 
[Backstage passes to a TAFKAL80ETC concert, 2, 7] => Backstage passes to a TAFKAL80ETC concert, 1, 10 
[Backstage passes to a TAFKAL80ETC concert, 2, 20] => Backstage passes to a TAFKAL80ETC concert, 1, 23 
[Backstage passes to a TAFKAL80ETC concert, 2, 49] => Backstage passes to a TAFKAL80ETC concert, 1, 50 
[Backstage passes to a TAFKAL80ETC concert, 2, 50] => Backstage passes to a TAFKAL80ETC concert, 1, 50 
[Backstage passes to a TAFKAL80ETC concert, 5, 0] => Backstage passes to a TAFKAL80ETC concert, 4, 3 
[Backstage passes to a TAFKAL80ETC concert, 5, 1] => Backstage passes to a TAFKAL80ETC concert, 4, 4 
[Backstage passes to a TAFKAL80ETC concert, 5, 7] => Backstage passes to a TAFKAL80ETC concert, 4, 10 
[Backstage passes to a TAFKAL80ETC concert, 5, 20] => Backstage passes to a TAFKAL80ETC concert, 4, 23 
[Backstage passes to a TAFKAL80ETC concert, 5, 49] => Backstage passes to a TAFKAL80ETC concert, 4, 50 
[Backstage passes to a TAFKAL80ETC concert, 5, 50] => Backstage passes to a TAFKAL80ETC concert, 4, 50 
[Backstage passes to a TAFKAL80ETC concert, 10, 0] => Backstage passes to a TAFKAL80ETC concert, 9, 2 
[Backstage passes to a TAFKAL80ETC concert, 10, 1] => Backstage passes to a TAFKAL80ETC concert, 9, 3 
[Backstage passes to a TAFKAL80ETC concert, 10, 7] => Backstage passes to a TAFKAL80ETC concert, 9, 9 
[Backstage passes to a TAFKAL80ETC concert, 10, 20] => Backstage passes to a TAFKAL80ETC concert, 9, 22 
[Backstage passes to a TAFKAL80ETC concert, 10, 49] => Backstage passes to a TAFKAL80ETC concert, 9, 50 
[Backstage passes to a TAFKAL80ETC concert, 10, 50] => Backstage passes to a TAFKAL80ETC concert, 9, 50 
[Backstage passes to a TAFKAL80ETC concert, 15, 0] => Backstage passes to a TAFKAL80ETC concert, 14, 1 
[Backstage passes to a TAFKAL80ETC concert, 15, 1] => Backstage passes to a TAFKAL80ETC concert, 14, 2 
[Backstage passes to a TAFKAL80ETC concert, 15, 7] => Backstage passes to a TAFKAL80ETC concert, 14, 8 
[Backstage passes to a TAFKAL80ETC concert, 15, 20] => Backstage passes to a TAFKAL80ETC concert, 14, 21 
[Backstage passes to a TAFKAL80ETC concert, 15, 49] => Backstage passes to a TAFKAL80ETC concert, 14, 50 
[Backstage passes to a TAFKAL80ETC concert, 15, 50] => Backstage passes to a TAFKAL80ETC concert, 14, 50 
[Conjured Mana Cake, -5, 0] => Conjured Mana Cake, -6, 0 
[Conjured Mana Cake, -5, 1] => Conjured Mana Cake, -6, 0 
[Conjured Mana Cake, -5, 7] => Conjured Mana Cake, -6, 3 
[Conjured Mana Cake, -5, 20] => Conjured Mana Cake, -6, 16 
[Conjured Mana Cake, -5, 49] => Conjured Mana Cake, -6, 45 
[Conjured Mana Cake, -5, 50] => Conjured Mana Cake, -6, 46 
[Conjured Mana Cake, -1, 0] => Conjured Mana Cake, -2, 0 
[Conjured Mana Cake, -1, 1] => Conjured Mana Cake, -2, 0 
[Conjured Mana Cake, -1, 7] => Conjured Mana Cake, -2, 3 
[Conjured Mana Cake, -1, 20] => Conjured Mana Cake, -2, 16 
[Conjured Mana Cake, -1, 49] => Conjured Mana Cake, -2, 45 
[Conjured Mana Cake, -1, 50] => Conjured Mana Cake, -2, 46 
[Conjured Mana Cake, 0, 0] => Conjured Mana Cake, -1, 0 
[Conjured Mana Cake, 0, 1] => Conjured Mana Cake, -1, 0 
[Conjured Mana Cake, 0, 7] => Conjured Mana Cake, -1, 3 
[Conjured Mana Cake, 0, 20] => Conjured Mana Cake, -1, 16 
[Conjured Mana Cake, 0, 49] => Conjured Mana Cake, -1, 45 
[Conjured Mana Cake, 0, 50] => Conjured Mana Cake, -1, 46 
[Conjured Mana Cake, 2, 0] => Conjured Mana Cake, 1, 0 
[Conjured Mana Cake, 2, 1] => Conjured Mana Cake, 1, 0 
[Conjured Mana Cake, 2, 7] => Conjured Mana Cake, 1, 5 
[Conjured Mana Cake, 2, 20] => Conjured Mana Cake, 1, 18 
[Conjured Mana Cake, 2, 49] => Conjured Mana Cake, 1, 47 
[Conjured Mana Cake, 2, 50] => Conjured Mana Cake, 1, 48 
[Conjured Mana Cake, 5, 0] => Conjured Mana Cake, 4, 0 
[Conjured Mana Cake, 5, 1] => Conjured Mana Cake, 4, 0 
[Conjured Mana Cake, 5, 7] => Conjured Mana Cake, 4, 5 
[Conjured Mana Cake, 5, 20] => Conjured Mana Cake, 4, 18 
[Conjured Mana Cake, 5, 49] => Conjured Mana Cake, 4, 47 
[Conjured Mana Cake, 5, 50] => Conjured Mana Cake, 4, 48 
[Conjured Mana Cake, 10, 0] => Conjured Mana Cake, 9, 0 
[Conjured Mana Cake, 10, 1] => Conjured Mana Cake, 9, 0 
[Conjured Mana Cake, 10, 7] => Conjured Mana Cake, 9, 5 
[Conjured Mana Cake, 10, 20] => Conjured Mana Cake, 9, 18 
[Conjured Mana Cake, 10, 49] => Conjured Mana Cake, 9, 47 
[Conjured Mana Cake, 10, 50] => Conjured Mana Cake, 9, 48 
[Conjured Mana Cake, 15, 0] => Conjured Mana Cake, 14, 0 
[Conjured Mana Cake, 15, 1] => Conjured Mana Cake, 14, 0 
[Conjured Mana Cake, 15, 7] => Conjured Mana Cake, 14, 5 
[Conjured Mana Cake, 15, 20] => Conjured Mana Cake, 14, 18 
[Conjured Mana Cake, 15, 49] => Conjured Mana Cake, 14, 47 
[Conjured Mana Cake, 15, 50] => Conjured Mana Cake, 14, 48 
